src/Admin/SubsiteAdmin.php

```diff
@@ -31,11 +39,18 @@ public function getEditForm($id = null, $fields = null)
     {
         $form = parent::getEditForm($id, $fields);
 
-        $grid = $form->Fields()->dataFieldByName(str_replace('\\', '-', Subsite::class));
-        if ($grid) {
-            $grid->getConfig()->getComponentByType(GridFieldPaginator::class)->setItemsPerPage(100);
-            $grid->getConfig()->removeComponentsByType(GridFieldDetailForm::class);
-            $grid->getConfig()->addComponent(new GridFieldSubsiteDetailForm());
+        if ($this->modelClass === Subsite::class) {
+            $grid = $form->Fields()->dataFieldByName(str_replace('\\', '-', Subsite::class));
+            if ($grid) {
+                $grid->getConfig()->getComponentByType(GridFieldPaginator::class)->setItemsPerPage(100);
+                $grid->getConfig()->removeComponentsByType(GridFieldDetailForm::class);
+                $grid->getConfig()->addComponent(new GridFieldSubsiteDetailForm());
+            }
+        }
+
+        if ($this->modelClass === SubsiteDomain::class) {
+            $grid = $form->Fields()->dataFieldByName($this->sanitiseClassName(SubsiteDomain::class));
```

Comment:
Would this `sanitiseClassName` call work for the Subsite::class branch above as well to keep it consistent?

Reply:
Yup. I was considering doing that but I wasn't sure if I should do it in this pull request since it's _technically_ not part of the same change. I'll make that change along with the other minor suggestions you've made :)